File: flutter/vex_owl_admin/lib/core/utils/api_client.dart
import 'package:dio/dio.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../constants/app_constants.dart';
import 'storage_utils.dart';

final dioProvider = Provider<Dio>((ref) {
  final dio = Dio(
    BaseOptions(
      baseUrl: AppConstants.apiBaseUrl,
      connectTimeout: AppConstants.httpTimeout,
      receiveTimeout: AppConstants.httpTimeout,
      headers: {
        'Content-Type': 'application/json',
      },
    ),
  );

  dio.interceptors.add(
    InterceptorsWrapper(
      onRequest: (options, handler) async {
        final token = await StorageUtils.getToken();
        if (token != null && token.isNotEmpty) {
          options.headers['Authorization'] = 'Bearer $token';
        }
        return handler.next(options);
      },
      onError: (error, handler) {
        if (error.response?.statusCode == 401) {
          StorageUtils.clearAuth();
        }
        return handler.next(error);
      },
    ),
  );

  return dio;
});

class ApiException implements Exception {
  final String message;
  final int? statusCode;

  ApiException(this.message, [this.statusCode]);

  @override
  String toString() => message;
}

class ApiClient {
  final Dio _dio;

  ApiClient(this._dio);

  Future<T> get<T>(
    String path, {
    Map<String, dynamic>? queryParameters,
    T Function(dynamic)? parser,
  }) async {
    try {
      final response = await _dio.get(path, queryParameters: queryParameters);
      final data = response.data;
      if (data['code'] == 0 || data['success'] == true) {
        final result = parser != null ? parser(data['data']) : data['data'] as T;
        return result;
      } else {
        throw ApiException(data['message'] ?? '请求失败');
      }
    } on DioException catch (e) {
      throw ApiException(
        e.response?.data?['message'] ?? e.message ?? '网络错误',
        e.response?.statusCode,
      );
    }
  }

  Future<T> post<T>(
    String path, {
    dynamic data,
    T Function(dynamic)? parser,
  }) async {
    try {
      final response = await _dio.post(path, data: data);
      final responseData = response.data;
      if (responseData['code'] == 0 || responseData['success'] == true) {
        final result = parser != null ? parser(responseData['data']) : responseData['data'] as T;
        return result;
      } else {
        throw ApiException(responseData['message'] ?? '请求失败');
      }
    } on DioException catch (e) {
      throw ApiException(
        e.response?.data?['message'] ?? e.message ?? '网络错误',
        e.response?.statusCode,
      );
    }
  }

  Future<T> put<T>(
    String path, {
    dynamic data,
    T Function(dynamic)? parser,
  }) async {
    try {
      final response = await _dio.put(path, data: data);
      final responseData = response.data;
      if (responseData['code'] == 0 || responseData['success'] == true) {
        final result = parser != null ? parser(responseData['data']) : responseData['data'] as T;
        return result;
      } else {
        throw ApiException(responseData['message'] ?? '请求失败');
      }
    } on DioException catch (e) {
      throw ApiException(
        e.response?.data?['message'] ?? e.message ?? '网络错误',
        e.response?.statusCode,
      );
    }
  }

  Future<void> delete(String path) async {
    try {
      final response = await _dio.delete(path);
      final data = response.data;
      if (data['code'] != 0 && data['success'] != true) {
        throw ApiException(data['message'] ?? '请求失败');
      }
    } on DioException catch (e) {
      throw ApiException(
        e.response?.data?['message'] ?? e.message ?? '网络错误',
        e.response?.statusCode,
      );
    }
  }
}

final apiClientProvider = Provider<ApiClient>((ref) {
  return ApiClient(ref.read(dioProvider));
});

File: owl-admin/src/core/utils/apiClient.ts
import axios, { AxiosInstance, AxiosError } from 'axios';
import { AppConstants } from '../constants/appConstants';
import { storage } from './storage';

export class ApiException extends Error {
  constructor(message: string, public statusCode?: number) {
    super(message);
    this.name = 'ApiException';
  }
}

interface ApiResponse<T> {
  code: number;
  data: T;
  message?: string;
  success?: boolean;
}

class ApiClient {
  private client: AxiosInstance;

  constructor() {
    this.client = axios.create({
      baseURL: AppConstants.apiBaseUrl,
      timeout: AppConstants.httpTimeout,
      headers: {
        'Content-Type': 'application/json',
      },
    });

    this.client.interceptors.request.use(async (config) => {
      const token = storage.getToken();
      if (token) {
        config.headers.Authorization = `Bearer ${token}`;
      }
      return config;
    });

    this.client.interceptors.response.use(
      (response) => response,
      (error: AxiosError) => {
        if (error.response?.status === 401) {
          storage.clearAuth();
          window.location.href = '/login';
        }
        return Promise.reject(error);
      }
    );
  }

  async get<T>(url: string, params?: Record<string, unknown>): Promise<T> {
    try {
      const response = await this.client.get<ApiResponse<T>>(url, { params });
      const data = response.data;
      if (data.code === 0 || data.success) {
        return data.data;
      }
      throw new ApiException(data.message || '请求失败');
    } catch (error) {
      if (error instanceof AxiosError) {
        const message = error.response?.data?.message || error.message || '网络错误';
        throw new ApiException(message, error.response?.status);
      }
      throw error;
    }
  }

  async post<T>(url: string, data?: unknown): Promise<T> {
    try {
      const response = await this.client.post<ApiResponse<T>>(url, data);
      const responseData = response.data;
      if (responseData.code === 0 || responseData.success) {
        return responseData.data;
      }
      throw new ApiException(responseData.message || '请求失败');
    } catch (error) {
      if (error instanceof AxiosError) {
        const message = error.response?.data?.message || error.message || '网络错误';
        throw new ApiException(message, error.response?.status);
      }
      throw error;
    }
  }

  async put<T>(url: string, data?: unknown): Promise<T> {
    try {
      const response = await this.client.put<ApiResponse<T>>(url, data);
      const responseData = response.data;
      if (responseData.code === 0 || responseData.success) {
        return responseData.data;
      }
      throw new ApiException(responseData.message || '请求失败');
    } catch (error) {
      if (error instanceof AxiosError) {
        const message = error.response?.data?.message || error.message || '网络错误';
        throw new ApiException(message, error.response?.status);
      }
      throw error;
    }
  }

  async delete<T>(url: string): Promise<T> {
    try {
      const response = await this.client.delete<ApiResponse<T>>(url);
      const responseData = response.data;
      if (responseData.code === 0 || responseData.success) {
        return responseData.data;
      }
      throw new ApiException(responseData.message || '请求失败');
    } catch (error) {
      if (error instanceof AxiosError) {
        const message = error.response?.data?.message || error.message || '网络错误';
        throw new ApiException(message, error.response?.status);
      }
      throw error;
    }
  }
}

export const apiClient = new ApiClient();